dvc/repo/__init__.py

```diff
@@ -155,7 +157,8 @@ def _ignore(self):
 
     def check_dag(self, stages):
         """Generate graph including the new stage to check for errors"""
-        self.graph(stages=stages)
+        self.reset()
```

Comment:
also, why do we need to reset it here? it probably means that it's not just a regular check, but actual update or recalculate graph. Is there a problem that we will have some discrepancy between the in-memory representation and what we have on disk for some while? What I want probably to ask is this - if our intention is to check then should we avoid changing the cached properties and use reset when it's actually need?